utils/makeappicons.sh

```diff
@@ -148,27 +148,27 @@ map() {
 }
 
 # Now do the actual work
-find $DSTDIR -type f -name "*.svg" | while read -r DST;do
+find "$DSTDIR" -type f -name "*.svg" | while read -r DST; do
 
     # Find the icon name (base name without extender)
-    NAME=$(basename $DST .svg)
+    NAME=$(basename "$DST" .svg)
 
     # Find the base name of the svg file for this icon
-    MAT=$(map $NAME)
-    if [[ -z $MAT ]];then
+    MAT=$(map "$NAME")
+    if [[ -z $MAT ]]; then
         echo -e "${YELLOW}Warning: No MaterialDesign mapping for ${NAME}${NC}"
         continue
     fi
 
     # So the source file is:
     SRC="$MATERIAL/svg/$MAT.svg"
-    if [ ! -f "$SRC" ];then
+    if [[ ! -f $SRC ]]; then
         echo -e "${RED}Error: Source for ${NAME} doesn't exist: ${SRC}${NC}"
         continue
     fi
 
     # Replace the icon file with the source file
-    cp "$SRC" "$DST" || exit
+    cp -- "$SRC" "$DST" || exit
```

Comment:
Why this change?

Reply:
The `--` is there because $SRC cannot be guaranteed to *not* start with a `-` and hence could be interpreted as an option to `cp`. Using `--` means 'end processing of option arguments'. It is likely an unnecessary and overly cautious change, sure, I can rebase without it.

Taken from Guideline 10 at https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html#tag_12_02

Reply:
Fair enough makes sense to me now